cpp/src/copying/contiguous_split.cu

```diff
@@ -0,0 +1,550 @@
+                                       in.offset() };
+      mutable_column_view  mcv{in.type(), in.size(), data, 
+                               validity, validity == nullptr ? UNKNOWN_NULL_COUNT : 0 };      
+      if(in.nullable()){         
+         // PERFORMANCE TEST 2, 3
+         // rmm::device_scalar<cudf::size_type> valid_count{0, 0, rmm::mr::get_default_resource()};
+         copy_in_place_kernel<block_size, T, true><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_hacked), 
+                           *mutable_column_device_view::create(mcv)
+                           // PERFORMANCE TEST 2, 3, 4
+                           //, valid_count.data()
+                           );
+         // PERFORMANCE TEST 2, 3, 4
+         // mcv.set_null_count(in.size() - valid_count.value());                  
+      } else
+       {
+         copy_in_place_kernel<block_size, T, false><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_hacked), 
+                           *mutable_column_device_view::create(mcv)
+                           // PERFORMANCE TEST 2, 3, 4
+                           /*, nullptr*/);
+      }
+      mcv.set_null_count(cudf::UNKNOWN_NULL_COUNT);
+
+      out_cols.push_back(mcv);
+   }
+};
+
+// TODO
+template<>
+struct column_copy_functor_impl<string_view> {
+   void operator()(column_view const& in, char *&dst, std::vector<column_view>& out_cols
+                  // PERFORMANCE TEST 4
+                  /*, rmm::device_scalar<cudf::size_type> &valid_count*/)
+   {       
+   };
+};
+
+struct column_copy_functor {
+   template<typename T>   
+   void operator()(column_view const& in, char *&dst, std::vector<column_view>& out_cols
+                  // PERFORMANCE TEST 4
+                  /*, rmm::device_scalar<cudf::size_type> &valid_count*/)
+   {
+      column_copy_functor_impl<T> fn{};      
+      fn(in, dst, out_cols 
+         // PERFORMANCE TEST 4
+         /* , valid_count*/);
+   }
+};
+
+struct contiguous_split_result {
+   cudf::table_view                    table;
+   std::unique_ptr<rmm::device_buffer> all_data;
+};
+typedef std::vector<cudf::column_view> subcolumns;
+
+std::vector<contiguous_split_result> contiguous_split(cudf::table_view const& input,
+                                                      std::vector<size_type> const& splits,
+                                                      rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                                      cudaStream_t stream = 0)
+{    
+   // slice each column into a set of sub-columns   
+   std::vector<subcolumns> split_table;
+   split_table.reserve(input.num_columns());
+   for(int idx=0; idx<input.num_columns(); idx++){
+      split_table.push_back(cudf::experimental::slice(input.column(idx), splits));
```

Comment:
This looks like it could be replace with the zero-copy `slice` API that already exists: https://github.com/rapidsai/cudf/blob/branch-0.12/cpp/include/cudf/copying.hpp#L325

Reply:
Furthermore, the zero-copy `slice` API should be extended to provide a `table_view` API to do a zero-copy slice of a table.

Reply:
Just to clarify, the above code is using the zero-copy slice.   You're saying :  make a zero-copy slice function that works on table views?    

Did you want it to be split or slice?  At the moment, split is just a wrapper around slice that takes a different form for the input.